Flaky-test: KafkaAuthorizationKafkaMultitenantTenantMetadataTest.testListTopic

<!--- 

Instructions for reporting a flaky test using this issue template:

1. Replace [test class] in title and body with the test class name
2. Replace [test method] in title and body with the test method that failed. Multiple methods are flaky, remove the content that refers to the test method.
3. Replace "url here" with a url to an example failure. In the Github Actions workflow run logs, you can right click on the line number to copy a link to the line. Example of such url is https://github.com/streamnative/kop/pull/802/checks?check_run_id=3856903052#step:5:322 . The logs are available for a limited amount of time (usually for a few weeks).
4. Replace "relevant parts of the exception stacktrace here" with the a few lines of the stack trace that shows at least the exception message and the line of test code where the stacktrace occurred.
5. Replace "full exception stacktrace here" with the full exception stacktrace from logs. This section will be hidden by default.
6. Remove all unused fields / content to unclutter the reported issue. Remove this comment too.

-->
KafkaAuthorizationKafkaMultitenantTenantMetadataTest is flaky. The testListTopic test method fails sporadically.

[example failure](https://github.com/streamnative/kop/runs/4210083340?check_suite_focus=true)

```
Error:  Tests run: 15, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 24.993 s <<< FAILURE! - in io.streamnative.pulsar.handlers.kop.KafkaAuthorizationKafkaMultitenantTenantMetadataTest
Error:  testListTopic(io.streamnative.pulsar.handlers.kop.KafkaAuthorizationKafkaMultitenantTenantMetadataTest)  Time elapsed: 0.078 s  <<< FAILURE!
java.lang.AssertionError: expected [2] but found [0]
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertEqualsImpl(Assert.java:137)
	at org.testng.Assert.assertEquals(Assert.java:118)
	at org.testng.Assert.assertEquals(Assert.java:652)
	at org.testng.Assert.assertEquals(Assert.java:662)
	at io.streamnative.pulsar.handlers.kop.KafkaAuthorizationTestBase.testListTopic(KafkaAuthorizationTestBase.java:337)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java
+++ b/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java
@@ -593,7 +593,12 @@ public class KafkaRequestHandler extends KafkaCommandDecoder {
                     pulsarTopicsFuture.complete(allTopicMap);
                     return;
                 }
-                AtomicInteger allTopicCount = new AtomicInteger(allTopicMap.size());
+                AtomicInteger allTopicCount =
+                        new AtomicInteger(allTopicMap.values().stream().mapToInt(List::size).sum());
+                if (allTopicCount.get() == 0) {
+                    pulsarTopicsFuture.complete(topicMap);
+                    return;
+                }
                 final Runnable completeOne = () -> {
                     if (allTopicCount.decrementAndGet() == 0) {
                         pulsarTopicsFuture.complete(topicMap);

